contracts/contracts/TrustedSubscription.sol

```diff
@@ -137,6 +137,24 @@ contract TrustedSubscription is Ownable {
     emit Subscribed(projectId, msg.sender, block.timestamp);
   }
 
+  function isSubscribed(uint256 projectId, uint16 tierIndex, address donator) view public returns (bool) {
```

Comment:
This function can be optimized - `subscriptionDetails` will hold tier id, tier will hold token id and therefore we can get subscription group from projectId and donator address only.
Also, `SubscriptionGroup` has `subscriptionIndices` variable so we can get subscription without iteration through all array.